server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -96,8 +96,11 @@ public class MetadataIndexTemplateService {
             Map.of("type", DateFieldMapper.CONTENT_TYPE, "ignore_malformed", "false")
         );
         try {
-            DEFAULT_TIMESTAMP_MAPPING = new CompressedXContent(
+            DEFAULT_TIMESTAMP_MAPPING_WITHOUT_ROUTING = new CompressedXContent(
                 (builder, params) -> builder.startObject(MapperService.SINGLE_MAPPING_NAME)
+                    .startObject(RoutingFieldMapper.NAME)
+                    .field("required", false)
+                    .endObject()
```

Comment:
I think this needs a comment as to why this is necessary

Reply:
I [commented](https://github.com/elastic/elasticsearch/pull/97317/files#r1285950199) elsewhere about this in the PR. Did you mean that, or adding a comment to the code?

Reply:
I meant in the code, so that future code spelunkers know why it's there.